oauth.go

```diff
@@ -935,14 +1053,19 @@ func (s *SHA1Signer) Verify(message string, signature string) error {
 	return nil
 }
 
-func (s *SHA1Signer) SignatureMethod() string {
-	return SIGNATURE_METHOD_HMAC_SHA1
+func (s *HMACSigner) SignatureMethod() string {
+	return SIGNATURE_METHOD_HMAC + HASH_METHOD_MAP[s.HashFunc()]
```

Comment:
This isn't a big deal; I'm going to merge this patch.  I'm just thinking out loud about possible future directions.

If someone were to use NewCustomConsumer they could pass in a crypto.Hash which isn't in HASH_METHOD_MAP.  If so, I think the map lookup will return an empty value: "" for strings. 

Of course the comments at the top of the file say we only support SHA1 and SHA256, but it might be nice to help people out if they mess this up.

Some ideas:
- Instead of just passing in a crypto.Hash pass a pair of (crypto.Hash, hash-name string).
- Panicing or returning an error here, or in the constructor function if we see an unsupported hash.

I like the first one, since it seems like it's the most flexible: in theory this library would now support any hash function.  However, I'm worried about exposing an interface that suggests we support any hash function without actually having tested it.